query-engine/connector-test-kit/src/test/scala/util/SchemaBaseV11.scala

```diff
@@ -214,8 +225,11 @@ trait SchemaBaseV11 extends PlayJsonExtensions {
                                     childId <- idOptions;
 
                                     // Based on Id and relation fields
-                                    childReference  <- childReferences(simple, parentId, onParent, onChild);
-                                    parentReference <- parentReferences(simple, childId, childReference, onParent, onChild);
+                                    childReferenceToParent  <- childReferences(simple, parentId, onParent, onChild);
+                                    parentReferenceToChild <- parentReferences(simple, childId, childReferenceToParent, onParent, onChild);
+
+                                    // skip required virtual relation fields as those are not allowed
+                                    if onParent.isRequired && parentReferenceToChild != noRef;
```

Comment:
Why don't we need to do this for `onChild` as well?

Reply:
My suspicion is you woult need it, but you explicitly removed all the cases where both required child and required parent were passed in. I think therefore you do not even need to change this code since all tests that would produce now invalid output got removed.